packages/contracts-bedrock/src/Safe/DeputyGuardianModule.sol

```diff
@@ -0,0 +1,112 @@
+    }
+
+    /// @notice Getter function for the Safe contract instance
+    /// @return safe_ The Safe contract instance
+    function safe() public view returns (Safe safe_) {
+        safe_ = SAFE;
+    }
+
+    /// @notice Getter function for the SuperchainConfig's address
+    /// @return superchainConfig_ The SuperchainConfig's address
+    function superchainConfig() public view returns (SuperchainConfig superchainConfig_) {
+        superchainConfig_ = SUPERCHAIN_CONFIG;
+    }
+
+    /// @notice Getter function for the deputy guardian's address
+    /// @return deputyGuardian_ The deputy guardian's address
+    function deputyGuardian() public view returns (address deputyGuardian_) {
+        deputyGuardian_ = DEPUTY_GUARDIAN;
+    }
+
+    function _onlyDeputyGuardian() internal view {
+        require(msg.sender == DEPUTY_GUARDIAN, "DeputyGuardianModule: Only the deputy guardian can call this function.");
+    }
+
+    /// @notice Calls the Security Council Safe's `execTransactionFromModuleReturnData()`, with the arguments
+    ///      necessary to call `pause()` on the `SuperchainConfig` contract.
+    ///      Only the deputy guardian can call this function.
+    function pause() external {
+        _onlyDeputyGuardian();
+        bytes memory data = abi.encodeCall(SUPERCHAIN_CONFIG.pause, (""));
+
+        (bool success, bytes memory returnData) =
+            SAFE.execTransactionFromModuleReturnData(address(SUPERCHAIN_CONFIG), 0, data, Enum.Operation.Call);
+        require(success, string(returnData));
+    }
+
+    /// @notice Calls the Security Council Safe's `execTransactionFromModuleReturnData()`, with the arguments
+    ///      necessary to call `unpause()` on the `SuperchainConfig` contract.
+    ///      Only the deputy guardian can call this function.
+    function unpause() external {
+        _onlyDeputyGuardian();
+        bytes memory data = abi.encodeCall(SUPERCHAIN_CONFIG.unpause, ());
+
+        (bool success, bytes memory returnData) =
+            SAFE.execTransactionFromModuleReturnData(address(SUPERCHAIN_CONFIG), 0, data, Enum.Operation.Call);
+        require(success, string(returnData));
+    }
+
+    /// @notice Calls the Security Council Safe's `execTransactionFromModuleReturnData()`, with the arguments
+    ///      necessary to call `blacklistDisputeGame()` on the `OptimismPortal2` contract.
+    ///      Only the deputy guardian can call this function.
+    /// @param _portal The `OptimismPortal2` contract instance.
+    /// @param _game The `IDisputeGame` contract instance.
+    function blacklistDisputeGame(OptimismPortal2 _portal, IDisputeGame _game) external {
+        _onlyDeputyGuardian();
+        bytes memory data = abi.encodeCall(OptimismPortal2.blacklistDisputeGame, (_game));
+
+        (bool success, bytes memory returnData) =
+            SAFE.execTransactionFromModuleReturnData(address(_portal), 0, data, Enum.Operation.Call);
+        require(success, string(returnData));
+    }
```

Comment:
This (and the following) function allows the Foundation to call any address, with limitations on the function selector and abi encoded args. 

I think this is safe, but appreciate input on that.

Reply:
IMO it would be nicer if the portal address were a constructor parameter here. I don't love the idea that a function naming clash can accidentally make things unsafe down the line.

Reply:
The trouble is that there are many possible portal addresses in the Superchain, and there is no source of truth for them on-chain.

Reply:
Hmmm maybe just have a mapping of allowed portal addresses that the Safe itself can update? Does that work?

Reply:
That would mean that an on-chain action is required by the Security Council anytime we want to add a new chain to the Superchain, which is a big process change from the current superchain-registry based approach. 

This module is already explicitly allowing the Foundation multisig to take some very significant actions, so it is being trusted by the Security Council to behave with the understanding that the module will be disabled if not. IMO allowing the Foundation to alos make calls to `address(x).setRespectedGameType()` is not a huge extra risk.

Reply:
My concern is more about function signature collisions than anything else, I think I would approve this if we had a check on the rest of the codebase that there is no function signature that collides with `setRespectedGameType()`.